rundeckapp/src/test/groovy/org/rundeck/app/providers/GormUserDataProviderSpec.groovy

```diff
@@ -50,17 +50,16 @@ class GormUserDataProviderSpec extends RundeckHibernateSpec implements DataTest
         }
 
         when:
-        provider.findOrCreateUser(caseSensitiveParam ? loginName : loginName.toUpperCase())
+        def rdUser =provider.findOrCreateUser(loginName.toUpperCase())
 
         then:
-        if (caseSensitiveParam) {
-            User.findAll().size() == 2
-        } else {
-            User.findAll().size() == 1
-        }
+        User.findAll().size() == userCountSpec
+        rdUser.login == userDataSpec
 
         where:
         caseSensitiveParam << [false, true]
+        userCountSpec << [1,2]
+        userDataSpec << ["loginName1","LOGINNAME1"]
```

Comment:
shouldn't this just be 
```suggestion
        userDataSpec = "LOGINNAME1"
```

Reply:
I'm validating loginName1 because, on the first test, the search is case-insensitive, and even if I'm searching for the user as LOGINNAME1 the method shouldn't create a new user instead should find the existing one that is loginName1 on the other case where the search is case-sensitive since there is no coincidence with that user the method will create a user LOGINNAME1. That's why I also added the userCountSpec, to validate that on the first search, the user is being found and on the second search, the user is created since it doesn't exist.